composer/algorithms/no_op_model/no_op_model.py

```diff
@@ -26,7 +26,7 @@ class NoOpModelClass(ComposerModel):
 
     def __init__(self, original_model: torch.nn.Module):
         super().__init__()
-        self.weights = torch.tensor([1.5], requires_grad=True, dtype=torch.float)
+        self.weights = torch.nn.Parameter(torch.Tensor([1.5]))
```

Comment:
I'm a bit confused on how this would fix a device error...would we need to set the device explicitly, somewhat like this?

```suggestion
        original_device = next(original_model.parameters()).device
        self.weights = torch.nn.Parameter(torch.Tensor([1.5]), device=original_device)
```

Reply:
no_op_model algorithm runs on the init event (https://github.com/mosaicml/composer/blob/dev/composer/trainer/trainer.py#L649), i.e., before moving the model parameters to gpu. So `original_device` in your code above is still cpu. Once we make it a parameter, module_to_device (https://github.com/mosaicml/composer/blob/dev/composer/trainer/trainer.py#L710) takes care of it.

Reply:
Ahhh, makes sense. Thanks!